shell.nix

```diff
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+	let
+	overrides = (builtins.fromTOML (builtins.readFile ./rust-toolchain.toml));
+	libPath = with pkgs; lib.makeLibraryPath [
+		# load external libraries that you need in your rust project here
+	];
+in
+pkgs.mkShell rec {
+	# Kanidm dependencies
+	buildInputs = with pkgs; [
+		pkg-config
+		
+		git
+		rustup
+
+		clang
+		llvmPackages.bintools
+		
+		openssl
+		systemd
+		linux-pam
+	];
+	
+	RUSTC_VERSION = overrides.toolchain.channel;
+	# https://github.com/rust-lang/rust-bindgen#environment-variables
+	LIBCLANG_PATH = pkgs.lib.makeLibraryPath [ pkgs.llvmPackages_latest.libclang.lib ];
+	shellHook = ''
+		export PATH=$PATH:''${CARGO_HOME:-~/.cargo}/bin
+		export PATH=$PATH:''${RUSTUP_HOME:-~/.rustup}/toolchains/$RUSTC_VERSION-x86_64-unknown-linux-gnu/bin/
```

Comment:
Does it really need this arch-specific path? That seems … bad

Reply:
That is probably pkgs.stdenv.targetPlatform.rust.rustcTarget but this is bad regardless, see comment about rustup

Reply:
aarch64-* is probably more popular than x86 at this point (at least nixpkgs has better support for it). But yeah

Reply:
This has been removed entirely